engine/audio/src/audiocapture.cpp

```diff
@@ -151,10 +153,10 @@ double AudioCapture::fillBandsData(int number)
     // representing all the frequencies from 0 to m_sampleRate Hz.
     // I will just consider 0 to 5000Hz and will calculate average magnitude
     // for the number of desired bands.
-    double maxMagnitude = 0;
+    double bandMagnitude = 0;
 #ifdef HAS_FFTW3
-    unsigned int i = 0;
-    int subBandWidth = ((m_captureSize * SPECTRUM_MAX_FREQUENCY) / m_sampleRate) / number;
+    unsigned int i = 1; // skip DC bin
+    int subBandWidth = ((m_captureSize * SPECTRUM_MAX_FREQUENCY) / m_sampleRate) / defaultBarsNumber();
 
     for (int b = 0; b < number; b++)
```

Comment:
I don't understand why this loop is here, surely you don't mean to compute the magnitude for all bands from zero up to the `number`eth band ?

Reply:
IIRC `number` is the amount of bars wanted by the user. The FFT is always performed on the whole spectrum but the signal strength visualization is based on sub-bands.
You can have up to 32. Default is 16.
So this is a mistake: you need to use `number`.

Reply:
Ok I see now